src/duct/__main__.py

```diff
@@ -91,6 +91,71 @@ def max(self, other: ProcessStats) -> ProcessStats:
         )
 
 
+@dataclass
+class Suffix:
+    stdout: str = "stdout"
+    stderr: str = "stderr"
+    usage: str = "usage.json"
+    info: str = "info.json"
+
+    def __iter__(self) -> Generator:
+        for each in fields(self):
+            yield each.name, getattr(self, each.name)
+
+
+@dataclass
+class LogPaths:
+    stdout: str
+    stderr: str
+    usage: str
+    info: str
+
+    def __iter__(self) -> Generator:
+        for each in fields(self):
+            yield each.name, getattr(self, each.name)
+
+    def __post_init__(self) -> None:
+        self._prefix = ""
+
+    @property
+    def prefix(self) -> str:
+        return self._prefix  # type: ignore
+
+    @classmethod
+    def create(cls, output_prefix: str, pid: None | int = None) -> LogPaths:
+        datetime_filesafe = datetime.now().strftime("%Y.%m.%dT%H.%M.%S")
+        formatted_prefix = output_prefix.format(
+            pid=pid, datetime_filesafe=datetime_filesafe
+        )
+        args_dict = {name: f"{formatted_prefix}{suffix}" for name, suffix in Suffix()}
+        new = cls(**args_dict)
+        new._prefix = formatted_prefix
+        return new
+
+    def prepare_paths(self, clobber: bool, capture_outputs: Outputs) -> None:
+        conflicts = [path for path in asdict(self).values() if Path(path).exists()]
+        if conflicts and not clobber:
+            raise FileExistsError(
+                "Conflicting files:\n"
+                + "\n".join(f"- {path}" for path in conflicts)
+                + "\nUse --clobber to overrwrite conflicting files."
+            )
+        elif conflicts and clobber:
+            for path in conflicts:
+                Path(path).unlink()
+
+        if self.prefix.endswith(os.sep):  # If it ends in "/" (for linux) treat as a dir
+            os.makedirs(self.prefix, exist_ok=True)
+        else:
+            # Path does not end with a separator, treat the last part as a filename
+            directory = os.path.dirname(self.prefix)
+            if directory:
+                os.makedirs(directory, exist_ok=True)
+        for path_name, path in self:
+            if path_name in capture_outputs or capture_outputs == Outputs.ALL:
```

Comment:
I find the use of `path_name in capture_outputs` here rather obtuse.  It only works because `Outputs` inherits `str`, and the fact that `"stdout" not in Outputs.ALL` is counter-intuitive at first glance.  There's also the fact that, once you realize we're working with strings, `==` would make more sense than `in`.

I would suggest giving `Outputs` a `has(self, stream_name: str) -> bool` method that encapsulates this line's logic but compares `stream_name` against `self.value` using `==`.